### ISSUE TITLE
LDD Calculation - On Hold Days Added

### ISSUE THREAD

**Issue by JHarrietha-AOT** [bcgov/foi-flow#3197]
**Describe the bug in current situation**
When a user has put a request on hold and taken it off hold then put it back on hold in the same day, the current calculation will add 1 additional day for each time a request is taken off hold. This leads to an incorrect LDD Calculation for our users and applicants.

LDD recalculation is happening when a request is off hold to Response.

**Link bug to the User Story**
#3142

**Impact of this bug**
High Impact, having an incorrect higher LDD could lead to complaints from applicants who pay close attention to the details of there requests

**Chance of Occurring (high/medium/low/very low)**
low-medium
it is unlikely an analyst would take a request on and off hold multiple times in a day. But if they did due to human error this would have a big impact

**Pre Conditions: which Env, any pre-requesites or assumptions to execute steps?**

**Steps to Reproduce**
have already spoken to @divyav-aot to explain the issues with the calculation in the backend

**Actual/ observed behaviour/ results**

**Expected behaviour**
LDD calculation should only be adding one day for a given day that the request is put on hold, and not stack this days on top of one another if it is moved back and forth multiple times in the same day.

LDD shouldn't get calculated when a request is off hold to response.

**Screenshots/ Visual Reference/ Source**
If applicable, add screenshots to help explain your problem. You an use screengrab.

**Comment by JHarrietha-AOT** [comment]
@KyEggleston @liseandtea Have a look and let me know if you miss anything

@divyav-aot Feel free to add an estimate if you have one

**Comment by sumathi-thirumani-aot** [comment]
@JHarrietha-AOT  In other words, The intention is to avoid due-date recalculation due to multiple payment attempts in a single day. Please confirm.

**Comment by divyav-aot** [comment]
@JHarrietha-AOT @liseandtea Few more queries related to LDD calculation.
1. The applicant pay fee estimate and the outstanding on the same day (Meaning: On Hold to CFR and On Hold to Response happening on the same day). The LDD gets calculated only once. That means, the new LDD gets calculated when the applicant pays the fee estimate. LDD will not get calculated again when the state change from On Hold to response(applicant pays outstanding balance) on the same as applicant pays the fee estimate. Please correct me if I am wrong. 
2. The comments with new LDD shows only when the applicant pay the full estimate. If the applicant pay the deposit or outstanding balance, then the comments just show like below. Do you want to add the new LDD as part of the deposit and outstanding balance comment
![image.png](https://images.zenhubusercontent.com/6081a4273965c56c02fdc82c/6bbc544a-fd87-423b-b6c7-4fce86145caa)
cc. @sumathi-thirumani-aot

**Comment by liseandtea** [comment]
@divyav-aot @sumathi-thirumani-aot @JHarrietha-AOT 

Hey!
1) Correct, we are not calculating a new LDD if the applicant is paying the fee estimate balance and the request is moving from On Hold state back to Response state. At that point the analyst would be releasing the records to the applicant and closing the file. A new LDD is only needed if the applicant pays the fee estimate in full or pays the deposit and the file is moving from On Hold state back to CFR state. 

2) Adding the new LDD to a comment is only needed if the applicant has paid the estimate in full or paid the deposit. There is no need to add it to the final payment comment since this is leading to the records being release and the file closing.

**Comment by liseandtea** [comment]
@divyav-aot 

No, we would not want a further calculation of LDD days when a request has reached the response state and the records are ready to release to the applicant and fees are owing on the request. If a request reaches response state and there is no fee estimate owing, the analyst would proceed to release the records and then close the file.  When there is a fee estimate, instead of releasing the records and closing the file, the analyst places the file on hold and waits for the applicant to pay the balance. Even though the file is still technically open, it is still considered to be a closed file because there are no further actions for IAO to take except release the records to the applicant once payment is made.

**Comment by divyav-aot** [comment]
Thanks @liseandtea . Just wanted to inform you that, currently we recalculate the LDD even in Onhold to Response transition. If this is the case then, we have a bug in production. We will remove the calculation as part of this bug.
cc. @sumathi-thirumani-aot

**Comment by divyav-aot** [comment]
Added above issue as part of this bug.

**Comment by sumathi-thirumani-aot** [comment]
Extended Changes in this list:
========================

1. Changes in comment to include due date for partial payment
2. Changes in notification to include due date for  partial payment
3. Changes to avoid re-calculation of LDD during On Hold to Response

**Comment by liseandtea** [comment]
@sumathi-thirumani-aot 

Morning! I tested this on EDU-2022-12347 all LDD's are working correctly. 

1. I created a new request with an opening date of December 19, 2022 and the original LDD was  February 2, 2023.
2. File was placed on hold December 22, 2022.
3. File was taken off hold on December 23, 2022 and the new LDD was calculated as February 6, 2023 (this is correct).
4. I placed the file on hold several times and the LDD remained the same and is no longer stacking (this is the desired result).
5. I moved the request through the stages and sent the applicant the balance due email.
6. Applicant paid the balance due and when the request moved back to response the LDD remained February 6, 2023.
7. Closed request!
![Screenshot 2022-12-23 082849.jpg](https://images.zenhubusercontent.com/633f19609b1468159fc7d8b5/b9ad5fa3-207f-4de5-a06e-627117a6c84b)
